### PR TITLE
Add purging of no longer allowed connections from conntrack

### PR DESCRIPTION
**Pull request by mati7337**
Previously adding a firewall rule didn't close already established connections: https://github.com/QubesOS/qubes-issues/issues/4141

### CODE REVIEW

**Review by marmarek — CHANGES_REQUESTED** [review]
Tests need an update. In existing tests you can just mock this call and ignore it, but add a test checking this functionality specifically - at the very least unit tests for functions you add.

File: qubesagent/firewall.py [hunk 1]
```diff
@@ -180,6 +181,85 @@ def update_handled(self, addr):
     def list_targets(self):
         return set(t.split('/')[2] for t in self.qdb.list('/qubes-firewall/'))
 
+    def conntrack_drop(self, src, con):
+        subprocess.Popen(['conntrack', '-D', '--src', src, '--dst', con[1],
```
[the remark this answers is not in the document]
Comment (mati7337) [reply qubesagent/firewall.py]: Changed it to `.run` as it doesn't raise an exception if exit code is non-zero. `.check_call` would raise an exception if connection dies in between `conntrack -L` and `conntrack -D`.

**Comment by mati7337** [comment]
I also changed `ip_network(rule[f'dst{family}'])` to `ip_network(rule[f'dst{family}'], False)` so it doesn't raise an exception if the host bits are set (`qvm-firewall` allows it).

**Comment by marmarek** [comment]
PipelineRetry

File: qubesagent/firewall.py [hunk 2]
```diff
@@ -180,6 +181,86 @@ def update_handled(self, addr):
...
+    def conntrack_get_connections(self, family, source):
+        connections = set()
+
+        with subprocess.Popen(['conntrack', '-L',
+                               '--family', f'ipv{family}', '--src', source],
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.STDOUT) as p:
+            while True:
+                line = p.stdout.readline()
+                print(line)
```
Comment (marmarek) [inline qubesagent/firewall.py:199]: This line wants removing, but I can do that post-merge.

**Review by marmarek — APPROVED** [review]

**Review by marmarek — APPROVED** [review]

**Pull request merged**